pkg/netpol/connlist/conns_formatter.go

```diff
@@ -67,3 +69,56 @@ func (j jsonFormatter) writeOutput(conns []Peer2PeerConnection) (string, error)
 	}
 	return string(jsonConns), nil
 }
+
+// dotFormatter: implements the connsFormatter interface for dot output format
+type dotFormatter struct {
+}
+
+// formats an edge line from a singleConnFields struct , to be used for dot graph
+func (d dotFormatter) getEdgeLine(c singleConnFields) string {
+	return fmt.Sprintf("\t%q -> %q [label=%q color=\"gold2\" fontcolor=\"darkgreen\"]\n", c.Src, c.Dst, c.ConnString)
+}
+
+// formats a peer line for dot graph
+func (d dotFormatter) getPeerLine(peer eval.Peer) string {
+	var peerColor string
+	if peer.IsPeerIPType() {
+		peerColor = "red2"
+	} else {
+		peerColor = "blue"
+	}
+	peerName := peer.String()
+	return fmt.Sprintf("\t%q [label=%q color=%q fontcolor=%q]\n", peerName, peerName, peerColor, peerColor)
+}
+
+// sorts the final dot graph lines
+func (d dotFormatter) sortGraphLines(graphLines string) string {
+	var graphLinesSorted sort.StringSlice = strings.Split(graphLines, "\n")
+	graphLinesSorted.Sort()
+	return strings.Join(graphLinesSorted, "\n")
+}
```

Comment:
why do these functions take `dotFormatter` as a receiver?

Reply:
I wanted to tell these are "internal" for `dotFormatter` use only - (thinking of it as a class) - my fault ':)